docs/tutorial.md

```diff
@@ -197,116 +200,112 @@ curl -i http://localhost:3000/api/commands/ \
-{"type":"SHOPPING_ITEM_CREATED","payload":{"id":"3","text":"Canned beans"},"aggregateId":"shopping-list-1","aggregateVersion":4,"timestamp":1542884875144,"aggregateIdAndVersion":"shopping-list-1:4","_id":"qvKCvnEOhVQrD7xJ"}
-{"type":"SHOPPING_ITEM_CREATED","payload":{"id":"4","text":"Paper towels"},"aggregateId":"shopping-list-1","aggregateVersion":5,"timestamp":1542884890484,"aggregateIdAndVersion":"shopping-list-1:5","_id":"YEnzkAlBjEqaLwQI"}
-
+```sh
+sqlite3 data/event-store.db
+sqlite> select * from events;
+40|0|1603273983433|shopping-list-1|1|SHOPPING_LIST_CREATED|{"name":"List 1"}
+147|0|1603274037313|shopping-list-1|2|SHOPPING_ITEM_CREATED|{"id":"1","text":"Milk"}
 ```
 
 <!-- prettier-ignore-end -->
 
-### Performing Validation
+### Input Validation
 
-Your application's write side currently does not perform any input validation. This results in the following issues:
+Your application's write side currently does not validate input data. This results in the following issues:
 
-- The command handlers do not check whether all required fields are provided in a command's payload.
+- The command handlers do not check whether all required fields are in a command's payload.
 - It is possible to create more then one shopping list with the same aggregate ID.
-- You can create items in a in a nonexistent shopping list.
+- You can create items in a nonexistent shopping list.
 
-You can overcome the first issue by adding simple checks to each command handler:
+To overcome the first issue, add checks at the beginning of each command handler:
 
-**common/aggregates/shopping_list.commands.js:**
+**common/aggregates/shopping_list.commands.js**
 
 ```js
 createShoppingList: (state, { payload: { name } }) => {
-  if (!name) throw new Error("name is required");
+  if (!name) throw new Error('The "name" field is required');
   ...
 },
 createShoppingItem: (state, { payload: { id, text } }) => {
-  if (!id) throw new Error('id is required')
-  if (!text) throw new Error('text is required')
+  if (!id) throw new Error('The "id" field is required')
+  if (!text) throw new Error('The "text" field is required')
   ...
 }
 ```
 
-To overcome the second and third issues, you can use the **aggregate state**. This state is assembled on the fly by an aggregate **projection** from previously created events. To add a projection to the ShoppingList aggregate, create a **shopping_list.projection.js** file in the **common/aggregates** folder and add the following code there:
-
-**common/aggregates/shopping_list.projection.js:**
+To overcome the second and third issues, you need to have an **aggregate state** object that keeps track of what shopping lists were already created. An aggregate **projection** can assemble such an object from previously created events with the same aggregate ID. To add a projection to the ShoppingList aggregate, create a **shopping_list.projection.js** file in the **common/aggregates** folder and add the following code to this file:
```

Comment:
I think, 'will assemble' or 'assembles' is more correct in this case: "An aggregate projection **will** assemble such an object from ..."
It should be clear that all the previous events with the same aggregateId will be processed by projection prior to command handling.

Reply:
This makes sense. Changed to "assembles"